### PR TITLE
ROX-10071: Add SAC for Count for directly scoped resources

### PR DESCRIPTION
**Pull request by janisz**
## Description

Add SAC for Count for directly scoped resources.

## Testing Performed

CI

### CODE REVIEW

**Review by md2119 — APPROVED** [review]

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 1]
```diff
@@ -410,6 +410,8 @@ func (s *storeImpl) UpsertMany(ctx context.Context, objs []*{{.Type}}) error {
 func (s *storeImpl) Count(ctx context.Context) (int, error) {
 	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.Count, "{{.TrimmedType}}")
 
+    var sacQueryFilter *v1.Query
```
Comment (md2119) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:413]: you do not need to declare here.

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 2]
```diff
@@ -419,14 +421,27 @@ func (s *storeImpl) Count(ctx context.Context) (int, error) {
...
+    scopeChecker := sac.GlobalAccessScopeChecker(ctx)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.ResourceWithAccess{
+		Resource: targetResource,
+		Access:   storage.Access_READ_ACCESS,
+	})
+	if err != nil {
+		return 0, err
+	}
+    {{- if .Obj.IsClusterScope }}
+    sacQueryFilter, err = sac.BuildClusterLevelSACQueryFilter(scopeTree)
```
Comment (md2119) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:434]: `:=`

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 3]
```diff
@@ -419,14 +421,27 @@ func (s *storeImpl) Count(ctx context.Context) (int, error) {
...
+    {{- else}}
+    sacQueryFilter, err = sac.BuildClusterNamespaceLevelSACQueryFilter(scopeTree)
```
Comment (md2119) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:436]: `:=`

**Review by rhybrillou — APPROVED** [review]

File: tools/generate-helpers/pg-table-bindings/store.go.tpl (hunk 1 above)
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:413]: if I don't declare it here, I'd need another branch to set it to nil

**Pull request merged**